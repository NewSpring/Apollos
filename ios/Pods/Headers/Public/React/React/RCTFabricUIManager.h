../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h