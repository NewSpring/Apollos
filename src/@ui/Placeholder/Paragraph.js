import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Placeholder from 'rn-placeholder';
import { times } from 'lodash';
import styled from '@ui/styled';
import { Line } from './Line';

const ParagraphLine = styled(({ theme }) => ({
  height: theme.helpers.rem(1),
  marginVertical: (theme.helpers.rem(1) - theme.helpers.verticalRhythm(1, 0.5)) / 2,
}), 'Placeholder.Paragraph.line')(Line);

export const Paragraph = ({
  lineNumber = 3,
  width = '100%',
  lastLineWidth = '75%',
  firstLineWidth = '85%',
  lineStyle,
  style,
}) => (
  <View style={style}>
    {times(lineNumber, (i) => {
      let lineWidth = width;
      if (i === 0 && firstLineWidth) lineWidth = firstLineWidth;
      if (i === lineNumber - 1 && lastLineWidth) lineWidth = lastLineWidth;
      return (<ParagraphLine key={i} width={lineWidth} style={lineStyle} />);
    })}
  </View>
);

Paragraph.propTypes = {
  lineNumber: PropTypes.number,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  lastLineWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  firstLineWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  lineStyle: PropTypes.any, // eslint-disable-line
  style: PropTypes.any, // eslint-disable-line
};

export default Placeholder.connect(Paragraph);
