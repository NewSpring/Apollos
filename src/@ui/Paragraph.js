import { compose } from 'recompose';
import { View } from 'react-native';

import { withPlaceholder, Paragraph as ParagraphPlaceholder } from '@ui/Placeholder';
import styled from '@ui/styled';

const Paragraph = compose(
  styled(({ theme }) => ({
    paddingVertical: theme.sizing.baseUnit / 2,
  })),
  withPlaceholder(ParagraphPlaceholder),
)(View);

export default Paragraph;
