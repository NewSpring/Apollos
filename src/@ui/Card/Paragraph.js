import React from 'react';
import PropTypes from 'prop-types';
import Placeholder from 'rn-placeholder';
import { compose, getContext } from 'recompose';

import styled from '@ui/styled';
import { withTheme } from '@ui/theme';
import { UIText } from '@ui/typography';

const StyledText = styled(({ theme }) => ({
  marginVertical: theme.sizing.baseUnit / 4,
  width: '100%',
}))(UIText);

const enhance = compose(
  getContext({ isLoading: PropTypes.bool }),
  withTheme(),
);

const Paragraph = enhance(({
  isLoading,
  theme,
  children,
  ...props
}) => (
  <Placeholder.Paragraph
    width={'100%'}
    lineNumber={3}
    textSize={theme.helpers.rem(1.4)}
    onReady={!isLoading}
  >
    <StyledText {...props}>{children}</StyledText>
  </Placeholder.Paragraph>
));

export default Paragraph;
