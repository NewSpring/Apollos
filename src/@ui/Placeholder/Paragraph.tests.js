import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '@ui/TestProviders';
import { Paragraph } from './Paragraph';

describe('The Paragraph placeholder', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Providers><Paragraph /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders multiple lines', () => {
    const tree = renderer.create(
      <Providers><Paragraph lineNumber={50} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('has varying width lines', () => {
    const tree = renderer.create(
      <Providers><Paragraph lastLineWidth={33} firstLineWidth={'23%'} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('accepts custom line styles', () => {
    const tree = renderer.create(
      <Providers><Paragraph lineStyle={{ height: 33 }} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('accepts custom styles', () => {
    const tree = renderer.create(
      <Providers><Paragraph style={{ backgroundColor: 'salmon' }} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
