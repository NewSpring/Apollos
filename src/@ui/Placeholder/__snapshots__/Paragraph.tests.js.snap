// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Paragraph placeholder accepts custom line styles 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 33,
        "marginVertical": 4,
        "width": "85%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 33,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 33,
        "marginVertical": 4,
        "width": "75%",
      }
    }
  />
</View>
`;

exports[`The Paragraph placeholder accepts custom styles 1`] = `
<View
  style={
    Object {
      "backgroundColor": "salmon",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "85%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "75%",
      }
    }
  />
</View>
`;

exports[`The Paragraph placeholder has varying width lines 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "23%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": 33,
      }
    }
  />
</View>
`;

exports[`The Paragraph placeholder renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "85%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "75%",
      }
    }
  />
</View>
`;

exports[`The Paragraph placeholder renders multiple lines 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "85%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#dddddd",
        "borderRadius": 6,
        "height": 18,
        "marginVertical": 4,
        "width": "75%",
      }
    }
  />
</View>
`;
